diff --git a/bazel/api_build_system.bzl b/bazel/api_build_system.bzl
index d0f588c..699ec47 100644
--- a/bazel/api_build_system.bzl
+++ b/bazel/api_build_system.bzl
@@ -1,6 +1,6 @@
 load("@com_envoyproxy_protoc_gen_validate//bazel:pgv_proto_library.bzl", "pgv_cc_proto_library")
 load("@com_github_grpc_grpc//bazel:python_rules.bzl", _py_proto_library = "py_proto_library")
-load("@io_bazel_rules_go//go:def.bzl", "go_test")
+#load("@io_bazel_rules_go//go:def.bzl", "go_test")
 load("@io_bazel_rules_go//proto:def.bzl", "go_grpc_library", "go_proto_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load(
@@ -134,11 +134,6 @@ def xds_cc_test(name, **kwargs):
         **kwargs
     )
 
-def xds_go_test(name, **kwargs):
-    go_test(
-        name = name,
-        **kwargs
-    )
 
 # Old names for backward compatibility.
 # TODO(roth): Remove these once all callers are migrated to the new names.
@@ -148,5 +143,3 @@ def udpa_proto_package(srcs = [], deps = [], has_services = False, visibility =
 def udpa_cc_test(name, **kwargs):
   xds_cc_test(name, **kwargs)
 
-def udpa_go_test(name, **kwargs):
-  xds_go_test(name, **kwargs)
