diff --git a/platform/eSDK_LogAPI_V2.1.10/eSDKLogAPI/eSDKLog.cpp b/platform/eSDK_LogAPI_V2.1.10/eSDKLogAPI/eSDKLog.cpp
index 0847954..d94fc6f 100644
--- a/platform/eSDK_LogAPI_V2.1.10/eSDKLogAPI/eSDKLog.cpp
+++ b/platform/eSDK_LogAPI_V2.1.10/eSDKLogAPI/eSDKLog.cpp
@@ -111,9 +111,9 @@ wchar_t *GetWcharFromChar(const char *char_str)
 }
 
 static std::mutex locker;
-static std::shared_ptr<spdlog::logger> runLogger;
-static std::shared_ptr<spdlog::logger> operationLogger;
-static std::shared_ptr<spdlog::logger> interfaceLogger;
+static std::shared_ptr<ds_spdlog::logger> runLogger;
+static std::shared_ptr<ds_spdlog::logger> operationLogger;
+static std::shared_ptr<ds_spdlog::logger> interfaceLogger;
 
 #ifdef WIN32
 bool eSDKLog::InitSPDLOG(const std::string& product, unsigned int logLevel[LOG_CATEGORY], const std::wstring& logPath, int mode) {
@@ -159,41 +159,41 @@ bool eSDKLog::InitSPDLOG(const std::string& product, unsigned int logLevel[LOG_C
 	}
 
 	try {
-		runLogger = spdlog::rotating_logger_mt(m_InstanceRunName, logPath + L"/" + instanceRunName_W + L".log"
+		runLogger = ds_spdlog::rotating_logger_mt(m_InstanceRunName, logPath + L"/" + instanceRunName_W + L".log"
 			, ConfigMgrInstance().GetLogSize_Run() * 1024, ConfigMgrInstance().GetLogNum_Run());
-		runLogger->set_level(spdlog::level::level_enum(spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Run()));
+		runLogger->set_level(ds_spdlog::level::level_enum(ds_spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Run()));
 		//设置日志格式为 时间（精确到毫秒） 线程号 日志名 日志级别 自定义信息
 		runLogger->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [tid:%t] [%n] [%l] %v");
 		initiateSuccessfully = true;
 	}
-	catch (const spdlog::spdlog_ex &ex) {
+	catch (const ds_spdlog::spdlog_ex &ex) {
 		initiateSuccessfully = false;
 	}
 
 	if (initiateSuccessfully) {
 		try {
-			operationLogger = spdlog::rotating_logger_mt(m_InstanceOperationName, logPath + L"/" + instanceOperationName_W + L".log"
+			operationLogger = ds_spdlog::rotating_logger_mt(m_InstanceOperationName, logPath + L"/" + instanceOperationName_W + L".log"
 				, ConfigMgrInstance().GetLogSize_Run() * 1024, ConfigMgrInstance().GetLogNum_Run());
-			operationLogger->set_level(spdlog::level::level_enum(spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Operation()));
+			operationLogger->set_level(ds_spdlog::level::level_enum(ds_spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Operation()));
 			//设置日志格式为 时间（精确到毫秒） 线程号 日志名 日志级别 自定义信息
 			operationLogger->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [tid:%t] [%n] [%l] %v");
 			initiateSuccessfully = true;
 		}
-		catch (const spdlog::spdlog_ex &ex) {
+		catch (const ds_spdlog::spdlog_ex &ex) {
 			initiateSuccessfully = false;
 		}
 	}
 
 	if (initiateSuccessfully) {
 		try {
-			interfaceLogger = spdlog::rotating_logger_mt(m_InstanceInterfaceName, logPath + L"/" + instanceInterfaceName_W + L".log"
+			interfaceLogger = ds_spdlog::rotating_logger_mt(m_InstanceInterfaceName, logPath + L"/" + instanceInterfaceName_W + L".log"
 				, ConfigMgrInstance().GetLogSize_Run() * 1024, ConfigMgrInstance().GetLogNum_Run());
-			interfaceLogger->set_level(spdlog::level::level_enum(spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Interface()));
+			interfaceLogger->set_level(ds_spdlog::level::level_enum(ds_spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Interface()));
 			//设置日志格式为 时间（精确到毫秒） 线程号 日志名 日志级别 自定义信息
 			interfaceLogger->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [tid:%t] [%n] [%l] %v");
 			initiateSuccessfully = true;
 		}
-		catch (const spdlog::spdlog_ex &ex) {
+		catch (const ds_spdlog::spdlog_ex &ex) {
 			initiateSuccessfully = false;
 		}
 	}
@@ -212,41 +212,41 @@ bool eSDKLog::InitSPDLOG(const std::string& product, unsigned int logLevel[LOG_C
 	m_InstanceRunName = product + LOG_RUN_INSTANCE;
 
 	try {
-		runLogger = spdlog::rotating_logger_mt(m_InstanceRunName, logPath + "/" + m_InstanceRunName + ".log"
+		runLogger = ds_spdlog::rotating_logger_mt(m_InstanceRunName, logPath + "/" + m_InstanceRunName + ".log"
 			, ConfigMgrInstance().GetLogSize_Run() * 1024, ConfigMgrInstance().GetLogNum_Run());
-		runLogger->set_level(spdlog::level::level_enum(spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Run()));
+		runLogger->set_level(ds_spdlog::level::level_enum(ds_spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Run()));
 		//设置日志格式为 时间（精确到毫秒） 线程号 日志名 日志级别 自定义信息
 		runLogger->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [tid:%t] [%n] [%l] %v");
 		initiateSuccessfully = true;
 	}
-	catch (const spdlog::spdlog_ex &ex) {
+	catch (const ds_spdlog::spdlog_ex &ex) {
 		initiateSuccessfully = false;
 	}
 
 	if (initiateSuccessfully) {
 		try {
-			operationLogger = spdlog::rotating_logger_mt(m_InstanceOperationName, logPath + "/" + m_InstanceOperationName + ".log"
+			operationLogger = ds_spdlog::rotating_logger_mt(m_InstanceOperationName, logPath + "/" + m_InstanceOperationName + ".log"
 				, ConfigMgrInstance().GetLogSize_Run() * 1024, ConfigMgrInstance().GetLogNum_Run());
-			operationLogger->set_level(spdlog::level::level_enum(spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Operation()));
+			operationLogger->set_level(ds_spdlog::level::level_enum(ds_spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Operation()));
 			//设置日志格式为 时间（精确到毫秒） 线程号 日志名 日志级别 自定义信息
 			operationLogger->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [tid:%t] [%n] [%l] %v");
 			initiateSuccessfully = true;
 		}
-		catch (const spdlog::spdlog_ex &ex) {
+		catch (const ds_spdlog::spdlog_ex &ex) {
 			initiateSuccessfully = false;
 		}
 	}
 
 	if (initiateSuccessfully) {
 		try {
-			interfaceLogger = spdlog::rotating_logger_mt(m_InstanceInterfaceName, logPath + "/" + m_InstanceInterfaceName + ".log"
+			interfaceLogger = ds_spdlog::rotating_logger_mt(m_InstanceInterfaceName, logPath + "/" + m_InstanceInterfaceName + ".log"
 				, ConfigMgrInstance().GetLogSize_Run() * 1024, ConfigMgrInstance().GetLogNum_Run());
-			interfaceLogger->set_level(spdlog::level::level_enum(spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Interface()));
+			interfaceLogger->set_level(ds_spdlog::level::level_enum(ds_spdlog::level::debug + ConfigMgrInstance().GetLogLevel_Interface()));
 			//设置日志格式为 时间（精确到毫秒） 线程号 日志名 日志级别 自定义信息
 			interfaceLogger->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [tid:%t] [%n] [%l] %v");
 			initiateSuccessfully = true;
 		}
-		catch (const spdlog::spdlog_ex &ex) {
+		catch (const ds_spdlog::spdlog_ex &ex) {
 			initiateSuccessfully = false;
 		}
 	}
